src/api/factories/ListingItemFactory.ts

```diff
@@ -21,33 +22,33 @@ export class ListingItemFactory {
      * @param ListingItemMessage
      * @param marketId
      *
-     * @returns {ListingItemMessage}
+     * @returns {ListingItemTemplatePostRequest}
      */
 
-    public async getMessage(data: ListingItemMessage, marketId: number): Promise<ListingItemMessage> {
+    public async getMessage(data: ListingItemTemplatePostRequest, marketId: number): Promise<ListingItemMessage> {
         return {
             hash: data.hash,
-            listingItemTemplateId: data.listingItemTemplateId,
+            listingItemTemplateId: data.id,
             marketId,
             information: {
```

Comment:
hash is fine, but ListingItemMessage doesnt contain marketId or listingItemTemplateId.

https://kewde.gitbooks.io/protocol/content/public-listing-format/extended/extended-listing.html